### DISCUSSION TITLE
I wonder if it is not necessary to change the key of the alchemy provider when switching the chain.

### DISCUSSION THREAD

**Discussion by siosio34** [wevm/wagmi#2805] [Q&A] [answered]
This may be a stupid question because I'm still lacking.

I am using AlchemyProvider, but on the Alchemy site, the API key per chain is 1:1 mapping.

If I support several chains of goerli, sepolia, and mainnet and change the chain within the project, shouldn't I change the Alchemy Provider's API key to the one appropriate for the chain?

**Comment by siosio34** [comment]
Currently, when configuring the AlchemyProvider, it appears that any chain uses the same API_KEY. However, in Alchemy, I know that API_KEY used for each chain must be separately issued and used. I wonder if that part is intended or not, if it can be modified.

**Reply by jxom** [reply, marked as answer]
At the moment, Alchemy API keys are interoperable across any chain. However, the Alchemy team has advised folks they shouldn’t do this as this behavior may change in the future. We are going to be dropping this behavior in wagmi v2.
